src/Microsoft.ComponentDetection.Detectors/gradle/GradleComponentDetector.cs

```diff
@@ -47,6 +64,9 @@ protected override Task OnFileFoundAsync(ProcessRequest processRequest, IDiction
 
     private void ParseLockfile(ISingleFileComponentRecorder singleFileComponentRecorder, IComponentStream file)
     {
+        // Buildscript and Settings lockfiles are always development dependencies
+        var lockfileIsDevDependency = file.Location.EndsWith("buildscript-gradle.lockfile") || file.Location.EndsWith("settings-gradle.lockfile");
```

Comment:
lockfile paths are configurable. The buildscript-gradle.lockfile in particular is not a default lock path.

Reply:
Unless I'm misreading something, buildscript-gradle.lockfile *is* a standard name, it is documented here https://docs.gradle.org/current/userguide/dependency_locking.html#lock_state_location_and_format